pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -114,15 +138,16 @@ func doSetup(sourceURL url.URL, usernameSecret, passwordSecret, tokenSecret stri
 		username = DefaultUsername
 	}
 
-	// Remove user/pw from the source url
-	overrideSourceURL := sourceURL
-	overrideSourceURL.User = nil
+	overrideSourceURL, err := overrideURLFn(&sourceURL, username, password)
```

Comment:
Given that having different logic based on the length of the password is insane :-), I'd rather keep it out of the main function logic to make it easier to remove in the future.